src/connection.ts

```diff
@@ -1982,6 +1993,11 @@ export type FetchMiddleware = (
   fetch: (modifiedUrl: string, modifiedOptions: any) => void,
 ) => void;
 
+export type ExternalProvider = (request: {
+  method: string;
+  params?: Array<any>;
+}) => Promise<void>;
```

Comment:
doesn't the provider return a JSON RPC response object (not void)?

Reply:
Good catch @kklash. I will verify that in the morning.